python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -216,8 +225,100 @@ def test_cyclic_lr_in_training_pipeline(self):
 
         assert np.all(np.isclose(lr_gold, lr_list))
 
+    def test_reduce_lr_on_plateau_in_training_pipeline(self):
```

Comment:
@mohazahran I think you should probably add a test for the callback function alone. Creating new tensorflow models has been very memory intensive on the autobuilds and we should avoid it if possible.